module_utils/certificate_lsr/providers/certmonger.py

```diff
@@ -233,6 +244,39 @@ def set_auto_renew(self, auto_renew):
         return []
 
     def _set_user_and_group_if_different(self):
+        if not self.module.params.get("booted"):
+            (owner, group, mode) = self._get_permissions()
+            if not any([owner, group, mode]):
+                return False
+            if owner:
+                self.command_systemd(
+                    [
+                        "chown",
+                        owner,
+                        self.certificate_file_path,
+                        self.certificate_key_path,
+                    ]
+                )
+            if group:
+                self.command_systemd(
+                    [
+                        "chgrp",
+                        group,
+                        self.certificate_file_path,
+                        self.certificate_key_path,
+                    ]
+                )
+            if mode:
+                self.command_systemd(
+                    [
+                        "chmod",
+                        str(mode),
```

Comment:
This might be problematic - it looks like the module `mode` parameter can take an `int` value?  If so:

```python
>>> mode = 0o640
>>> mode
416
>>> str(mode)
'416'
```

I think either `mode` will need to be pre-formatted into an octal string like `"0640"`, or this code will need some additional logic to parse whatever type `mode` is in and format as desired.

Reply:
Ah right -- we only test it with strings like "0640" in tests/tests_fs_attrs.yml, but it is declared as `mode=dict(type="raw")`. But indeed the [set_fs_attributes_if_different() implementation](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L865) supports both.

Fixed.